etc/shadow-maint/userdel-pre.d/01-kill_user_procs.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+# Check user exists, and if so, send sigkill to processes that the user owns
+
+RUNNING=`ps -eo user | grep -Fx "$SUBJECT" | wc -l`
+
+# if the user does not exist, RUNNING will be 0
+
+if [ "${RUNNING}x" = "0x" ]; then
+  exit 0
+fi
+
+ls -1 /proc | while IFS= read -r PROC; do
+  if [ -d "/proc/${PROC}" ]; then
+    USR=`ls -ld "/proc/${PROC}" | awk '{print $3}'`
```

Comment:
What do you think about using 'stat -c "%U" /proc/${PROC}' instead of ls -ld | awk?  busybox (at least here) seems to do %U correctly.